openidconnect/lib/src/openidconnect_client.dart

```diff
@@ -368,18 +405,21 @@ class OpenIdConnectClient {
 
       return true;
     } on Exception catch (e) {
-      if (this._identity != null) {
-        await OpenIdIdentity.clear();
-        this._identity = null;
-
-        _raiseEvent(AuthEvent(AuthEventTypes.Error, message: e.toString()));
-      }
+      clearIdentity();
+      _raiseEvent(AuthEvent(AuthEventTypes.Error, message: e.toString()));
       return false;
     } finally {
       _refreshing = false;
     }
   }
 
+  void clearIdentity() async {
```

Comment:
i think this should be a Future<void>

Reply:
@ahmednfwela thanks for the feedback, it is fixed !